cirq-core/cirq/ops/gateset.py

```diff
@@ -324,6 +324,9 @@ def __contains__(self, item: Union[raw_types.Gate, raw_types.Operation]) -> bool
         g = item if isinstance(item, raw_types.Gate) else item.gate
         assert g is not None, f'`item`: {item} must be a gate or have a valid `item.gate`'
 
+        if isinstance(g, global_phase_op.GlobalPhaseGate):
+            return self._accept_global_phase_op
+
```

Comment:
Now that global phase op is a gate, we should remove the explicit `accept_global_phase_op` flag from gatesets and users should instead just insert `cirq.GlobalPhaseGate` in the gateset(`cirq.Gateset(cirq.GlobalPhaseGate, ...)` ) to accept the global phase gate. 

The parameter would also go through a deprecation cycle where for now, if the parameter is True we'll update the list of input gates to include `cirq.GlobalPhaseGate`. 

Is it possible to do this in a separate PR though? I suspect the tests would complain if we continue to use GlobalPhaseOperation in the Gateset for validation for now?

Reply:
Agreed, I plan to do that in a subsequent PR

Reply:
Can you please open an issue to track this?

Reply:
Done